### PR TITLE
Allow passing undefined options to move()

### LINKED ISSUE #947: Allow passing undefined options to move()
- **Operating System:**macos
- **Node.js version:**14.18.3
- **`fs-extra` version:**10.0.1

The follow code:

```
const { move } = require("fs-extra");

move("/some/path/that/exists", "some/path/that/doesnt/exist", undefined)
  .then(() => console.log("success")
  .catch(console.error);
```

produces the following type error: `TypeError: Cannot read property 'overwrite' of undefined`

The error seems to be from the handling of `optional callback` and `optional options` on line 12-15 of `lib/move/move.js`

### PR DESCRIPTION
**Pull request by Nokel81**
fixes #947 

I thought about adding a test but then couldn't find any tests of a similar nature within the repo so I didn't. I can if it is thought to be useful.

### CODE REVIEW

File: lib/move/move.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ function move (src, dest, opts, cb) {
     opts = {}
   }
 
+  opts = opts || {};
```
Comment (RyanZim) [inline lib/move/move.js:17]:
```suggestion
  opts = opts || {}
```

Comment (RyanZim) [inline lib/move/move.js:17]:
```suggestion
  opts = opts || {}
```

**Comment by Nokel81** [comment]
I have fixed the implementation to work with the existing tests and decided to add a unit test to cover this use case as well.

File: lib/move/__tests__/move.test.js [hunk 2]
```diff
@@ -331,6 +331,25 @@ describe('+ move()', () => {
...
+
+      fse.move(SRC_DIR, DEST_DIR, undefined)
+        .then(() => done())
+        .catch(error => {
+          if (error instanceof TypeError) {
+            done()
+          } else {
+            done(error)
+          }
+        })
```
Comment (RyanZim) [inline lib/move/__tests__/move.test.js:349]: Currently, the test would pass if a `TypeError` was thrown; fix:
```suggestion
        .catch(done)
```

**Comment by RyanZim** [comment]
Tests are failing; haven't looked into it in detail.

**Comment by RyanZim** [comment]
Created https://github.com/jprichardson/node-fs-extra/pull/955; which doesn't have the extra parameter validation, and has simpler tests.

**Pull request closed without merging**